src/mv_memory.rs

```diff
@@ -9,31 +9,36 @@ use crate::{
     ReadOrigin, ReadSet, TxIdx, TxVersion, WriteSet,
 };
 
-#[derive(Default)]
+#[derive(Default, Debug)]
 struct LastLocations {
     read: ReadSet,
     // Consider [SmallVec] since most transactions explicitly write to 2 locations!
     write: Vec<MemoryLocationHash>,
 }
 
+#[derive(Debug)]
 pub(crate) struct LazyAddresses(pub(crate) AHashSet<Address, BuildAddressHasher>);
 impl Default for LazyAddresses {
     fn default() -> Self {
         LazyAddresses(AHashSet::with_hasher(BuildAddressHasher::default()))
     }
 }
 
-// The MvMemory contains shared memory in a form of a multi-version data
-// structure for values written and read by different transactions. It stores
-// multiple writes for each memory location, along with a value and an associated
-// version of a corresponding transaction.
-pub(crate) struct MvMemory {
+/// The MvMemory contains shared memory in a form of a multi-version data
+/// structure for values written and read by different transactions. It stores
+/// multiple writes for each memory location, along with a value and an associated
+/// version of a corresponding transaction.
+#[derive(Debug)]
+pub struct MvMemory {
+    /// History of each location
```

Comment:
`The list of transaction incarnations and written values for each memory location`